zorder/src/main/scala/org/locationtech/sfcurve/zorder/Z2.scala

```diff
@@ -31,12 +32,11 @@ class Z2(val z: Long) extends AnyVal {
   def d1 = dim(1)
 
   def mid(p: Z2): Z2 = {
-    var ans: Z2  = new Z2(0)
-    if (p.z < z)
-      ans  = new Z2(p.z + (z - p.z)/2)
-    else
-      ans = new Z2(z + (p.z - z)/2)
-    ans
+    if (p.z < z) {
```

Comment:
ha, this is what i get for fixing the formatting :)
so the z value only uses 62 bits and ignores the sign flag. However, you're right I don't think that is accounted for in some of these methods.
I'm going to update it to use the same method as Z3, which is the mid of the decoded values and not the z values - i'll also change to max-value safe mean.